### PR TITLE
Add extrapolation for `sct_get_centerline` fitseg

### LINKED ISSUE #3531: Wrong behaviour of `fitseg` algorithm: the output centerline is not extrapolated
### Description

Usage says:

~~~
  -method {optic,viewer,fitseg}
                        Method used for extracting the centerline.
                          - optic: automatic spinal cord detection method
                          - viewer: manual selection a few points followed by interpolation
                          - fitseg: fit a regularized centerline on an already-existing cord segmentation. It will
                          interpolate if slices are missing and extrapolate beyond the segmentation boundaries (i.e., every
                          axial slice will exhibit a centerline pixel). (default: optic)
~~~

So, we expect the fitseg method to extrapolate the centerline _outside_ of the segmentation zmin/zmax, but it does not:

image

### Steps to Reproduce

0. Install SCT 666237aeb31246ff163905c7f3cf6cbe8d696e44

1. Download: [sag_cervical_seg.nii.gz](https://github.com/spinalcordtoolbox/spinalcordtoolbox/files/7282226/sag_cervical_seg.nii.gz)

2. Run:
   ~~~
   sct_get_centerline -i sag_cervical_seg.nii.gz -method fitseg -o sag_cervical_centerline
   ~~~

### Another minor issue

The usage should mention "(default: optic)" before the bullet point, not at the end of the fitseg description.

### PR DESCRIPTION
**Pull request by sandrinebedard**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/geek-tips/contributing#pr-labels-a-href-pr-labels-id-pr-labels-a) to this PR
- [x] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [ ] I've assigned a reviewer

#### PR contents

- [x] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Testing) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description
This PR adresses #3531. It adds the argument `-extrapolation` to extrapolate the centerline when using the method `fitseg`. 

## Linked issues
Resolves #3531

### CODE REVIEW

File: spinalcordtoolbox/scripts/sct_get_centerline.py [hunk 1]
```diff
@@ -79,6 +79,14 @@ def get_parser():
         default=0,
         help="Binary or soft centerline. 0 = binarized, 1 = soft. Only relevant with -method fitseg."
     )
+    optional.add_argument(
+        "-extrapolation",
+        metavar=Metavar.int,
+        type=int,
+        choices=[0, 1],
+        default=0,
+        help="Extrapolate beyond the segmentation boundaries. 0 = no extrapolation, 1 = extrapolation. Only relevant with -method fitseg."
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/scripts/sct_get_centerline.py:88]: I would perhaps consider adding a note about the potentially erratic behavior when combined with higher-order interpolation methods (bspline, polyfit), as described in https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3531#issuecomment-1483347201

In other words, we could mention that `-extrapolate` works best with lower-order interpolation methods?

File: spinalcordtoolbox/scripts/sct_get_centerline.py [hunk 2]
```diff
@@ -153,13 +161,17 @@ def main(argv: Sequence[str]):
...
+        printv("ERROR: -extrapolation can only be used with '-method fitseg'.", type='error')
+
     # Gap between slices
     interslice_gap = arguments.gap
 
     param_centerline = ParamCenterline(
         algo_fitting=arguments.centerline_algo,
         smooth=arguments.centerline_smooth,
-        minmax=True,
+        minmax=arguments.extrapolation,
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/scripts/sct_get_centerline.py:174]: Before, the default was `True`. Now, we use the default of `arguments.extrapolation`, which is `0`, which == `False`.

Is this change intentional? Or do we essentially need to reverse the value?

- `extrapolation=0` --> `minmax=True`
- `extrapolation=1` --> `minmax=False`

An easy way to do this could be to add a `not` here:

```suggestion
        minmax=(not arguments.extrapolation),  # NB: 'extrapolation=0' --> 'minmax=True' and vice versa
```

```python
>>> extrapolation = 0
>>> minmax = (not extrapolation)
>>> print(minmax)
True
```

**Review by joshuacwnewton — CHANGES_REQUESTED** [review]
I had only some quick comments here, since this is a straightforward addition! :blush:

File: spinalcordtoolbox/scripts/sct_get_centerline.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def get_parser():
              "  - viewer: manual selection a few points followed by interpolation\n"
              "  - fitseg: fit a regularized centerline on an already-existing cord segmentation. It will "
              "interpolate if slices are missing and extrapolate beyond the segmentation boundaries (i.e., every "
-             "axial slice will exhibit a centerline pixel)."
+             "axial slice will exhibit a centerline pixel) if -extrapolation 1 is specified."
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/scripts/sct_get_centerline.py:59]: I have one slight wording change:

```python
            " - fitseg: fit a regularized centerline on an already-existing cord segmentation. This method "
            "will interpolate if any slices are missing. Also, if  '-extrapolation 1` is specified, this method will "
            "extrapolate beyond the segmentation boundaries (i.e., every axial slice will exhibit a centerline pixel)."
```

Since this is a long sentence, it might be easy to miss the `-extrapolation 1` part, because it is at the end of the sentence. (People might get lazy and stop reading, hehe.)

So, my suggestion is to split the sentence into two, then add the `-extrapolate 1` part at the start of the sentence.

Comment (sandrinebedard) [reply spinalcordtoolbox/scripts/sct_get_centerline.py]: done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4079/commits/6959b7bfcf4c8419cc7758c214dbe1208e81f5e3

File: spinalcordtoolbox/scripts/sct_get_centerline.py [hunk 4]
```diff
@@ -86,6 +86,7 @@ def get_parser():
         choices=[0, 1],
         default=0,
         help="Extrapolate beyond the segmentation boundaries. 0 = no extrapolation, 1 = extrapolation. Only relevant with -method fitseg."
+             "Note: '-extrapolation 1' works best with lower-order centerline fitting algorithms"
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/scripts/sct_get_centerline.py:89]: Tiny tiny nitpick: It might be good to add in brackets which methods are lower-order? (Is it just linear? Or linear and nurbs?)

(I mention this because right now, the `-centerline-algo` flag doesn't contain any info on the various algorithms, so the user might not know what "lower-order" means.)

**Review by joshuacwnewton — APPROVED** [review]
I just had one last tiny nitpick, but other than that, this LGTM!

File: spinalcordtoolbox/scripts/sct_get_centerline.py (hunk 4 above)
Comment (sandrinebedard) [reply spinalcordtoolbox/scripts/sct_get_centerline.py]: nurbs too! done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4079/commits/483e72f0d9a8edc933b1629410c7eae5e81a83f0
